atlasdb-client/src/main/java/com/palantir/atlasdb/keyvalue/impl/AbstractKeyValueService.java

```diff
@@ -76,6 +79,7 @@
             + " while doing a write to {}. Attempting to batch anyways.";
 
     protected ExecutorService executor;
+    protected KeyValueServiceArgSupplier argSupplier = KeyValueServiceArgSupplier.NO_OP;
```

Comment:
OK. I've made this `volatile` for now (the object proper is immutable so I don't see any significant problems here).